File: packages/core/src/factories/internal/metadata/MetadataHelper.ts
import { MetadataConstant } from "../../../schemas/metadata/MetadataConstant";
import { MetadataConstantValue } from "../../../schemas/metadata/MetadataConstantValue";
import { MetadataSchema } from "../../../schemas/metadata/MetadataSchema";

export namespace MetadataHelper {
  export const literal_to_metadata = (key: string): MetadataSchema => {
    const metadata: MetadataSchema = MetadataSchema.initialize();
    metadata.constants.push(
      MetadataConstant.create({
        type: "string",
        values: [
          MetadataConstantValue.create({
            value: key,
            tags: [],
          }),
        ],
      }),
    );
    return metadata;
  };
}

File: packages/core/src/factories/internal/metadata/MetadataHelper.go
package metadata

type metadataHelper struct{}

var MetadataHelper metadataHelper

func (metadataHelper) LiteralToMetadata(key string) *MetadataSchema {
	metadata := NewMetadataSchema()
	metadata.Sole = key
	metadata.Constants = append(metadata.Constants, MetadataConstant{
		Type: "string",
		Values: []MetadataConstantValue{{
			Value: key,
			Tags:  [][]MetadataTypeTag{},
		}},
	})
	return metadata
}
